src/libs/actions/Report.js

```diff
@@ -224,6 +257,12 @@ function updateReportWithNewAction(reportID, reportAction) {
         maxSequenceNumber: reportAction.sequenceNumber,
     });
 
+    if (reportAction.clientID) {
+        // Remove the temporary action from the report since we are about to
```

Comment:
This comment refers to the "temporary" action, but it calls the method to remove a "local" action. Does the comment need updated or the method?

Also, why does the old one need to be removed if it's just going to get replaced? That essentially creates two writes to Onyx when only a single write is needed.

Reply:
`removeLocalAction()` can remove any local action that we have a key for. In this case, it's a temporary action we are removing.

The old one needs to be removed because it does not have the same key as the action we are adding here. We use the `clientID` as the key/sequenceNumber for the temporary. The "real" action will just use its true `sequenceNumber` as a key.

Reply:
You're right about the two writes though .. right on!

Maybe we can do a single merge here instead and just do

```js
const actionsToMerge = {};
if (reportAction.clientID) {
    actionsToMerge[reportAction.clientID] = null;
}
actionsToMerge[reportAction.sequenceNumber] = {...};
Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, actionsToMerge);
```

Reply:
Oh, I like that, yeah.